### PR TITLE
Fix Issue 18365 - header file generation doesn't include the return attribute

### PR DESCRIPTION
**Pull request by wilzbach**
This is a first approach to the bug that's blocking https://github.com/dlang/phobos/pull/6123.
I'm not a huge fan of working around against the existing code and I will look into accepting `return` as postfix argument too, but maybe someone has a better idea?

### CODE REVIEW

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -963,6 +969,9 @@ public:
             buf.writeByte(')');
         }
         parametersToBuffer(t.parameters, t.varargs);
+        buf.writeByte(' ');
```
Comment (RazvanN7) [inline src/dmd/hdrgen.d]: This should be inside the if statement, otherwise you will have a space even if you don't have a `return` attribut. That's what makes the tests to fail.

File: test/compilable/extra-files/header18365.d [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module foo.bar.ba;
+nothrow pure @nogc @safe package(foo) 
+{
+	void foo();
+	nothrow pure @nogc @safe package(foo.bar) void foo2();
+}
```
Comment (RazvanN7) [inline test/compilable/extra-files/header18365.d]: why is this necessary? You can add your test cases in the compilable/extra-files/header1.d or header2.d or header3.d.
There is no need to add an extra file.

**Review by RazvanN7 — DISMISSED** [review]
Looks good, except for the space error.

File: test/compilable/extra-files/header18365.d (hunk 2 above)
Comment (wilzbach) [reply test/compilable/extra-files/header18365.d]: Because of `module foo.bar.ba;`

**Comment by ZombineDev** [comment]
@wilzbach you need to fix the tests

**Comment by RazvanN7** [comment]
ping @wilzbach . What's the status on this?

**Comment by wilzbach** [comment]
Ping @RazvanN7 @JinShil @ZombineDev  - this is read now and finally passing all CIs.

File: test/runnable/testscope2.d [hunk 3]
```diff
@@ -42,7 +42,7 @@ void test3()
         assert(SS.foo4.mangleof == "_D10testscope22SS4foo4MFNcNkKNgPiZi");
 
         // Test scope pretty-printing
-        assert(typeof(SS.foo1).stringof == "ref return ulong(return ref int* delegate() return p)");
+        assert(typeof(SS.foo1).stringof == "ref ulong(return ref int* delegate() return p) return");
```
Comment (jacob-carlborg) [inline test/runnable/testscope2.d]: I hope this will not break downstream projects.

**Review by JinShil — APPROVED** [review]
